ADAL/src/request/ADDrsDiscoveryRequest.m

```diff
@@ -56,11 +56,16 @@ + (void)requestDrsDiscoveryForDomain:(NSString *)domain
 
 + (NSURL *)urlForDrsDiscoveryForDomain:(NSString *)domain adfsType:(AdfsType)type
 {
-    NSString *urlString = (type == AD_ADFS_ON_PREMS) ?
-    [NSString stringWithFormat:@"https://enterpriseregistration.%@/enrollmentserver/contract?api-version=1.0", domain.lowercaseString] :
-    [NSString stringWithFormat:@"https://enterpriseregistration.windows.net/%@/enrollmentserver/contract?api-version=1.0", domain.lowercaseString];
-    
-    return [NSURL URLWithString:urlString];
+    if (type == AD_ADFS_ON_PREMS)
+    {
+        return [NSURL URLWithString:
+                [NSString stringWithFormat:@"https://enterpriseregistration.%@/enrollmentserver/contract?api-version=1.0", domain.lowercaseString]];
+    }
+    else
```

Comment:
Make this a else if (AD_....) { ... } else { @throw "unrecognized type"; } Just in case...